GeneralsMD/Code/GameEngine/Source/GameClient/GUI/ControlBar/ControlBar.cpp

```diff
@@ -174,6 +174,14 @@ Relationship ControlBar::getCurrentlyViewedPlayerRelationship(const Team* team)
 	return NEUTRAL;
 }
 
+AsciiString ControlBar::getCurrentlyViewedPlayerSide()
+{
+	if (Player* player = getCurrentlyViewedPlayer())
```

Comment:
I do wonder, is the null test necessary at all? Would `return getCurrentlyViewedPlayer()->getSide()` be enough? Because `getLocalPlayer` is not supposed to return null.

Same with the function above. I remember in the previous change your logic in `getCurrentlyViewedPlayerRelationship` was different for NEUTRAL. Does the Observer have NEUTRAL relation with player teams?

Maybe these 2 additional functions are not necessary at all if we just need to call `getCurrentlyViewedPlayer()->getRelationship()`, `getCurrentlyViewedPlayer()->getSide()`